### ISSUE TITLE
Building core solar leads to stuck builders

### ISSUE THREAD

**Issue by derdodga** [beyond-all-reason/Beyond-All-Reason#757]
When building core solar panels, builders (especially commander) get stuck because they stand to close to the construction site. As soon as construction is started, the unit can only be freed by reclaiming the solar.

**Comment by Beherith** [comment]
Does this mainly happen with the first solar that is built by the commander?

**Comment by derdodga** [comment]
That is correct. It seems the building unit does not recognize the boundaries of the solar correctly and does not step aside far enough.

**Comment by KyleAnthonyShepherd** [comment]
![image](https://github.com/beyond-all-reason/Beyond-All-Reason/assets/44480662/45475e61-e30c-42a0-b105-9dc0e0d1a9f3)
Issue still exists

**Comment by KyleAnthonyShepherd** [comment]
![image](https://github.com/beyond-all-reason/Beyond-All-Reason/assets/44480662/e801216d-1159-4fa5-bbfc-8c94acc5248a)
Issue still exists

**Comment by SethDGamre** [comment]
Requesting review session to determine if this is something that can be fixed on the beyond all reason side or on the engine side.

**Comment by efrec** [comment]
This is still possible to do during pregame placement with space + build orders. That consists of two problems:

1. Before the initial spawn, `gui_pregame_build` is in control of avoiding commander/build clashes. Only the first building is not supposed to clash (though even then clashing is possible with shift), so further placements are not restricted.
2. Shift + space skips the whole clash check. The test in `DoBuildingsClash` in that file probably should be checked in most or all cases, instead (and should use `0.5 * Game.squareSize`).

Even with the above fixed, you can easily enclose a builder in its own solars. BAR has several ways of walling your own cons. That's mostly one problem, actually, I think:

1. The commander's initial position and map square match its initial location at its first slow update, [so `AMoveType::UpdateGroundBlockMap` noops out](https://github.com/beyond-all-reason/RecoilEngine/blob/83a8ead722af9b6ef83d30c664d98abc0806c7fc/rts/Sim/MoveTypes/MoveType.cpp#L99-L117).
2. There's no non-hack way to make that not happen. (A) Toggle the unit's blocking off and on again. (B) We can force an update by [attaching/detaching the commander](https://github.com/beyond-all-reason/RecoilEngine/blob/83a8ead722af9b6ef83d30c664d98abc0806c7fc/rts/Sim/Units/Unit.cpp#L787) (C) We can ask nicely for a way to call `AddGroundBlockingObject` directly or whatever the preferred flavor. (D) Builders that place a building without moving to do so should clear their old slow-update position and current map square or else be exempt from those checks.

**Comment by WatchTheFort** [comment]
Do we know what the actual root cause of this issue is? Queueing up a building on top of the Commander's current position is not a problem except for frame 1 on game start, so we shouldn't even need to do anything special to make it work.

**Comment by efrec** [comment]
The commander does not block its ground square until it has moved.

**Comment by WatchTheFort** [comment]
Why? Is this just something inherent to the engine?

**Comment by efrec** [comment]
It does not seem like a hard requirement of the engine, but it is the engine behavior.
